lib/LMSManagers/LMSHelpdeskManager.php

```diff
@@ -1718,8 +1718,13 @@ public function TicketChange($ticketid, array $props)
             $props['customerid'] = null;
         }
 
-        if (isset($props['categories'])) {
-            $ticket['categories'] = empty($ticket['categories']) ? array() : array_keys($ticket['categories']);
+        $ticket['categories'] = is_array($ticket['categories']) ? Utils::array_column($ticket['categories'], 'id') : null;
+        $props['categories'] = is_array($props['categories']) ? array_values($props['categories']) : null;
+
+        if ($props['categories'] !== $ticket['categories']) {
```

Comment:
Takie porównanie tablic wydaje się nieco ryzykowne:

```sh
php -r '$a = array(1, 2, 3); $b = array(2, 1, 3); var_dump($a !== $b);'
```

generuje:

```
bool(true)
```

Reply:
Są dwie opcje:
a) sortujemy obie tablice,
b) stosujemy `===` zamiast `==`

Która wg Ciebie lepsza?

Reply:
Żadna z tych opcji - zobacz jak gdzie indziej robione są porównania tablic.

Reply:
`!empty(array_diff($props['categories'], $ticket['categories'])` ?

Reply:
Kod nie jest podatny na duplikaty występujące w tablicy oraz zamianę miejsc elementów w tablicy. 

Testowałem go tak:
https://gist.github.com/interduo/f02ce4e0ce6d220d43aaac7dc1b0bbed

Posłużyłem się przykładem: https://github.com/chilek/lms/blob/99aa25c77582ffb4d0881bd1824126efbf333f59/lib/LMSManagers/LMSNetworkManager.php#L1037-L1039 ;-)

Jak masz jakiś lepszy/prostszy pomysł to zasugeruj.

Reply:
Jeśli taki sposób jest akceptowalny można by pokusić się o zrobienie `Utill::Compare_arrays()` co Ty na to?

Reply:
A array_diff() za długie jest? :)

Reply:
A `array_intersect()`? :)

Reply:
Czy masz na myśli taki test:
`empty(array_diff(array_intersect($array1, $array2), $array1)) && array_diff(array_intersect($array1, $array2), $array2)))`
?

Reply:
Może tak:
```php
if (count(array_intersect($array1, $array2)) == count($array1)) {
  ...
}
```

Reply:
Być może nawet to zadziała:
```php
if (array_intersect($array1, $array2) === $array1) {
  ...
}
```